lib/commands/dialog/index.js

```diff
@@ -68,51 +68,58 @@ class DialogCommand extends AbstractCommand {
      * @param {Object} cmd encapsulates arguments provided to the dialog command.
      * @return { skillId, locale, stage, profile, debug, replayFile, smapiClient, userInputs <for replay mode> }
      */
-    _getDialogConfig(cmd) {
+    _getDialogConfig(cmd, callback) {
         const debug = Boolean(cmd.debug);
         let { skillId, locale, stage, profile } = cmd;
         profile = profileHelper.runtimeProfile(profile);
         stage = stage || CONSTANTS.SKILL.STAGE.DEVELOPMENT;
         let firstLocaleFromManifest;
         let userInputs;
+        const smapiClient = new SmapiClient({ profile, doDebug: debug });
         if (cmd.replay) {
             const dialogReplayConfig = new DialogReplayFile(cmd.replay);
             skillId = dialogReplayConfig.getSkillId();
             if (!stringUtils.isNonBlankString(skillId)) {
-                throw 'Replay file must contain skillId';
+                return callback(new CliError('Replay file must contain skillId'));
```

Comment:
To keep the function asynchronous, this return has to be wrapped with `process.nextTick() => {}`